improver_tests/utilities/test_cube_extraction.py

```diff
@@ -483,5 +488,191 @@ def test_single_threshold_use_original_units(self):
+
+    def test_error_no_grid_spec(self):
+        """Test error when required grid_spec is not provided"""
+        msg = "grid_spec required"
+        with self.assertRaisesRegex(ValueError, msg):
+            subset_data(self.uk_cube)
+
+    def test_error_no_sites_available(self):
+        """Test error raised when none of the required sites are present in
+        the input spot cube"""
+        msg = "Cube does not contain any of the required sites"
+        with self.assertRaisesRegex(ValueError, msg):
+            subset_data(self.spot_cube, site_list=["3100"])
+
+    def test_error_wrong_grid_spec(self):
+        """Test error raised when required coordinates are not in the grid
+        spec dictionary"""
+        grid_spec_latlon = {
+            "longitude": {"min": 0, "max": 7, "thin": 2},
+            "latitude": {"min": 42, "max": 52, "thin": 2},
+        }
+        msg = "Cube coordinates .* are not present"
+        with self.assertRaisesRegex(ValueError, msg):
+            subset_data(self.uk_cube, grid_spec=grid_spec_latlon)
+
+    def test_error_non_overlapping_grid_spec(self):
+        """Test error raised when defined cutout does not overlap with input
+        cube"""
+        grid_spec_xy = {
+            "projection_x_coordinate": {"min": -10000, "max": -6000, "thin": 3},
+            "projection_y_coordinate": {"min": 0, "max": 10000, "thin": 3},
+        }
+        msg = "Cube domain does not overlap with cutout specified"
+        with self.assertRaisesRegex(ValueError, msg):
+            subset_data(self.uk_cube, grid_spec=grid_spec_xy)
+
+    def test_error_order_agnostic(self):
+        """Test non-overlapping grid error is correctly raised regardless of
+        which coordinate is non-overlapping. This is needed because there are
+        ways of calling iris.extract that appear to modify the input constraints,
+        such that the output of this function was not always correct depending on
+        which coordinate did not match."""
+        grid_spec_xy = {
+            "projection_y_coordinate": {"min": -10000, "max": -6000, "thin": 3},
+            "projection_x_coordinate": {"min": 0, "max": 10000, "thin": 3},
+        }
+        msg = "Cube domain does not overlap with cutout specified"
+        with self.assertRaisesRegex(ValueError, msg):
+            subset_data(self.uk_cube, grid_spec=grid_spec_xy)
+
+    def test_error_single_point(self):
+        """Function does not support extraction of a single (non-dimensioned)
+        point from a grid - test a suitable error is raised if this is requested"""
+        grid_spec = {
+            "projection_x_coordinate": {"min": 0, "max": 1000, "thin": 1},
+            "projection_y_coordinate": {"min": 0, "max": 1000, "thin": 1},
+        }
+        msg = "Function does not support single point extraction"
+        with self.assertRaisesRegex(ValueError, msg):
+            subset_data(self.uk_cube, grid_spec=grid_spec)
+
```

Comment:
I'd also like a test to check that for an x-y ordered input grid I get back an x-y ordered subset once you have made that the case.